eng/script-tests/Repo-File-Content-Replacements/Repo-File-Content-Replacements.Tests.ps1

```diff
@@ -0,0 +1,60 @@
+<#
+```
+Invoke-Pester -Output Detailed $PSScriptRoot/Repo-File-Content-Replacements.Tests.ps1
+```
+#>
+
+Import-Module Pester
+Set-StrictMode -Version Latest
+
+BeforeAll {
+    function Backup-Folder($targetFolder, $backupFolder) {
+        if (!(Test-Path $targetFolder)) {
+            return $null
+        }
+        
+        if (!(Test-Path $backupFolder)) {
+            New-Item $backupFolder -ItemType Directory
+        }
+        Copy-Item -Path "$targetFolder/*" -Destination $backupFolder -Recurse -PassThru -Force
+    }
+    function Reset-Folder($backupFolder, $targetFolder) {
+        if (Test-Path $backupFolder) {
+            Copy-Item -Path "$backupFolder/*" -Destination $targetFolder -Recurse -Force
+            Remove-Item $backupFolder -Recurse
+        }
+    }
+}
+AfterAll {
+    if (Test-Path "$PSScriptRoot/inputs/backup") {
+        Remove-Item "$PSScriptRoot/inputs/backup" -Recurse
+    }
+}
+# Test plan:
+# 1. Tests on the path does not match exclude nor include.
+# 1. Tests on the path matching exclude paths.
+# 2. Tests on the path matching include paths.
+# 3. Tests on the path matching both exclude and include.
+# 4. Tests on the file match the migration map.
+# 5. Tests on the file does not match the migration map.
+Describe "repo-file-content-replacement" -Tag "UnitTest" {
+    # Passed cases
+    It "Test on the files have matching content" -TestCases @(
+        @{ exludePaths = "(/|\\)excludes(/|\\)|(/|\\)eng(/|\\)scripts-tests(/|\\)Repo-File-Content-Replacements(/|\\)inputs(/|\\)test1(/|\\)jsonFiles(/|\\)1.json"; includePaths = "(/|\\)excludes(/|\\)includes(/|\\)"; migrationMapFile = "$PSScriptRoot/inputs/test1/jsonFiles/1.json"; scannedDirectory = "$PSScriptRoot/inputs/test1" }
+    ) {
+        $backupFolder = "$PSScriptRoot/inputs/backup"
+        Backup-Folder -targetFolder $ScannedDirectory -backupFolder $backupFolder
+        $migrationMap = Get-Content $migrationMapFile -Raw
+        . $PSScriptRoot/../../scripts/Repo-File-Content-Replacements.ps1 `
```

Comment:
Can you invoke this directly instead of dot sourcing it?

Reply:
The script is not under the same layer as test script. This probably the safest way to call the script. Do you have any better way?

Reply:
You can place 

`        . $PSScriptRoot/../../scripts/Repo-File-Content-Replacements.ps1 ``

[into the BeforeAll](https://github.com/Azure/azure-sdk-tools/blob/a67b73170db3591b16d0c38c7df3772f18fa1e24/tools/asset-sync/assets.Tests.ps1#L2). Saves needing to put it per line, makes it more "environmental".